marketplacetests/tests/test_marketplace_create_confirm_pin.py

```diff
@@ -13,22 +13,18 @@ class TestMarketplaceCreateConfirmPin(MarketplaceGaiaTestCase):
 
     def test_create_confirm_pin(self):
 
-        app_name = 'Test Zippy With Me'
         pin = '1234'
-        acct = FxATestAccount(base_url=self.base_url).create_account()
-
-        if self.apps.is_app_installed(app_name):
-            raise Exception('The app %s is already installed.' % app_name)
 
         marketplace = Marketplace(self.marionette, self.MARKETPLACE_DEV_NAME)
         home_page = marketplace.launch()
 
-        home_page.login(acct.email, acct.password)
+        app_name = home_page.find_first_paid_app()
```

Comment:
I was not really happy with this either, but it seemed necessary based on everything else that was happening behinf the scenes (e.g., checking for the app to be installed and logging in). Your comment made me realize that I missed something, which is that just because marionette loses its context doesn't mean we have to reload the page - we just have to find the element again!  I am now working on a better version of this PR which I will submit for another review shortly.